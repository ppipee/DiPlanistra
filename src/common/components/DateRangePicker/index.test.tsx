import React from 'react'

import { render } from '@testing-library/react'

import 'jest-styled-components'
import * as DateRangePickerType from '.'

describe('Test DateRangePicker', () => {
	const MOCK_DATE = 'Mock Date' as any
	const setDateSpy = jest.fn()
	const rangePickerPanelSpy = jest.fn()
	const dateInputSpy = jest.fn()
	const dateRangeProviderSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/components/DateRangeProvider', () => (props: any) => {
		dateRangeProviderSpy(props)
		return <div {...props}>{props.children}</div>
	})

	jest.doMock('./components/DateInput', () => (props: any) => {
		dateInputSpy(props)
		return <div {...props}>{props.children}</div>
	})

	jest.doMock('./components/RangePickerPanel', () => (props: any) => {
		rangePickerPanelSpy(props)
		return <div {...props}>{props.children}</div>
	})

	const { default: DateRangePicker } = require('.') as typeof DateRangePickerType

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render DateRangePicker correctly', () => {
		const { container } = render(<DateRangePicker date={MOCK_DATE} setDate={setDateSpy} />)

		dateRangeProviderSpy.mock.calls[0][0].setDate()
		expect(dateRangeProviderSpy.mock.calls[0][0].date).toBe(MOCK_DATE)
		expect(setDateSpy).toBeCalledTimes(1)

		expect(container.firstChild).toMatchSnapshot()
	})
})
