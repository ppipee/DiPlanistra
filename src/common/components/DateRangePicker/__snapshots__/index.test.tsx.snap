// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateRangePicker should render DateRangePicker correctly 1`] = `
<div
  date="Mock Date"
>
  <div />
</div>
`;
